src/round.rs

```diff
@@ -332,6 +335,8 @@ impl<Id, H, N, Signature> Round<Id, H, N, Signature> where
 						chain,
 					)?;
 
+					self.historical_votes.seen.push(Message::Prevote(vote.clone()));
+
 					None
 				}
 				VoteMultiplicity::Equivocated(ref first, ref second) => {
```

Comment:
Don't we need to store equivocated votes as well?

Reply:
At the very least, the first equivocation

Reply:
Fixed.